### ISSUE TITLE
<!-- eslint-disable --> does not work in the v1.0.0-beta.6

### ISSUE THREAD

**Issue by vsemozhetbyt** [eslint/eslint-plugin-markdown#69]
1\. I've found an error on the v1.0.0-beta.4: `<!-- eslint-disable some-rule -->` completely disables parsing the whole fragment and even the next fragment:

`test.md`
```md
### Heading

Text

` ``js
console.log('a')

` ``

` ``js
console.log('b')

` ``

` ``js
console.log('c')

` ``
```
Error log:
```console
   6:1   error  Use the global form of 'use strict'     strict
   6:17  error  Missing semicolon                       semi
  11:1   error  Use the global form of 'use strict'     strict
  11:17  error  Missing semicolon                       semi
  16:1   error  Use the global form of 'use strict'     strict
  16:17  error  Missing semicolon                       semi
```

`test.md` with comments:
```md

### Heading

Text

<!-- eslint-disable strict -->
` ``js
console.log('a')

` ``

` ``js
console.log('b')

` ``

` ``js
console.log('c')

` ``
```
Error log (`semi` errors for the fragments 1 and 2 are not reported; fragment 2 is ignored):
```
  17:1   error  Use the global form of 'use strict'     strict
  17:17  error  Missing semicolon                       semi
```

2\. So I've tried to update to v1.0.0-beta.6 and make a PR for Node.js. v1.0.0-beta.6 seems to fix this issue, but it has a new breaking one: code blocks with `<!-- eslint-disable -->` are parsed and throws if parsing errors found:
`test.md`:
```md
### Heading

Text

<!-- eslint-disable -->
` ``js
console.log('a') bang

` ``
```
Error log:
```console
  7:18  error  Parsing error: Unexpected token bang
```

**Comment by btmills** [comment]
@vsemozhetbyt thanks for reporting these!

1. Disable comments fall through

    Yikes! That's not a good bug. Based on a quick bisect, it looks like it was fixed in d2f29620f2e2204f009ae97d45d5b3b1b206dbb3. Strangely, I had [an empty section of tests](https://github.com/eslint/eslint-plugin-markdown/blob/65e296d71ce5ccd3b598ff1aa4c4ae9fb56442e7/tests/lib/plugin.js#L97-L99) where that should've been tested, but I guess I never wrote the tests. I've filed #70, which adds tests to make sure config comments don't fall through to subsequent blocks.

2. `eslint-disable` still reports parser errors

    This looks like it's behaving as intended, though maybe not as expected. This has come up [a](https://github.com/eslint/eslint/issues/5005) [handlful](https://github.com/eslint/eslint/issues/6869) [of](https://github.com/eslint/eslint/issues/900) [times](https://github.com/eslint/eslint/issues/8419) in the [eslint/eslint](https://github.com/eslint/eslint) issue tracker, so we added a note to [the docs about `eslint-disable` comments](http://eslint.org/docs/user-guide/configuring#disabling-rules-with-inline-comments):

    > Comments that disable warnings for a portion of a file tell ESLint not to report rule violations for the disabled code. ESLint still parses the entire file, however, so disabled code still needs to be syntactically valid JavaScript.

    While it would be technically feasible to override that behavior in the preprocessor, I'd like to keep this plugin inline with ESLint's intended behavior.

    What use case do you have where you have JS code fences (` ```js ... ``` `) that don't contain valid JS? Maybe there's a way to support it still. For example, this plugin only checks code fences that are [explicitly marked](https://github.com/eslint/eslint-plugin-markdown/blob/65e296d71ce5ccd3b598ff1aa4c4ae9fb56442e7/lib/processor.js#L12) as JS syntax. If the code isn't valid JS, are you able to change the code fence tag to something else?

**Comment by vsemozhetbyt** [comment]
@btmills 
We have started to use eslint-plugin-markdown in Node.js CI and local builds from the https://github.com/nodejs/node/pull/12563 with the v1.0.0-beta.4. We have many code blocks marked with the `<!-- eslint-disable -->` and they was not parsed in v1.0.0-beta.4, so we cannot update now without revisiting all these fragments.

Use cases examples for docs: REPL sessions logs that are not valid JS but need syntax highlighting; big Object listing that erroneously parsed as code blocks by ESLint (wrapping them in parentheses make them less readable. It would be very handy to have an option to disable eslint-plugin-markdown for such fragments: code fragments in docs do have some specificity.

**Comment by btmills** [comment]
@vsemozhetbyt that makes sense, thanks for explaining.

As a first step, I tracked down what was causing the behavior you observed in beta.4. It turns out that the Markdown parser in use at the time wasn't picking up code fences unless they were preceded by blank lines. I've added a test in #72 to make sure that doesn't happen again.

    

    ```js
    // This block is seen by both beta.4 and beta.6
    ```

    
    ```js
    // This block is invisible to beta.4 but seen by beta.6
    ```

Since your disable comments and code fences were on adjacent lines, the processor wasn't even getting those code blocks back from the parser. I'm glad that bug was fixed in a beta! 😅 

Now, as for supporting your use case. I'd like to keep the meaning of `eslint-disable` the same in ESLint core and this processor, as I fear it could introduce confusion otherwise. With that in mind, how about I add support for a new `<!-- eslint-skip -->` comment that would exclude the code fence from parsing entirely? If that works for you, I'll run it by the rest of the team to make sure I'm not stepping on any toes with that. It would look like this:

    
    ```js
    !@#$%^&*() <-- Doesn't even get parsed. The processor ignores this block entirely.
    ```

    
    ```js
    alert('Hello, ' + "world!") // Only check for syntax errors, ignore rules
    ```

**Comment by vsemozhetbyt** [comment]
@btmills I like it)

**Comment by Trott** [comment]
> @btmills I like it) Let's see what others think.

Me too.

**Comment by refack** [comment]
> With that in mind, how about I add support for a new 

👍

**Comment by btmills** [comment]
I just submitted a pull request to implement `<!-- eslint-skip -->` comments in Markdown: https://github.com/eslint/eslint-plugin-markdown/pull/73. Let me know what you think!

**Comment by vsemozhetbyt** [comment]
@btmills Thank you! Looking for landing and the new version to update the plugin in the Node.js.

**Comment by btmills** [comment]
`<!-- eslint-skip -->` comments have been released in [v1.0.0-beta.7](https://github.com/eslint/eslint-plugin-markdown/releases/tag/v1.0.0-beta.7).

**Comment by vsemozhetbyt** [comment]
@btmills Thank you!
